plugins/catalog-react/src/components/EntityRefLink/EntityRefLink.tsx

```diff
@@ -39,6 +61,107 @@ export type EntityRefLinkProps = {
   children?: React.ReactNode;
 } & Omit<LinkProps, 'to'>;
 
+type PeekAheadPopoverProps = {
+  popupState: PopupState;
+  entityRef: CompoundEntityRef;
+};
+
+const useStyles = makeStyles(() => {
+  return {
+    popoverPaper: {
+      width: '30em',
+    },
+  };
+});
+
+export const PeekAheadPopover = ({
+  popupState,
+  entityRef,
+}: PeekAheadPopoverProps) => {
+  const entityRoute = useRouteRef(entityRouteRef);
+  const classes = useStyles();
+  const apiHolder = useApiHolder();
+
+  const [{ loading, error, value: entity }, load] = useAsyncFn(async () => {
+    const catalogApi = apiHolder.get(catalogApiRef);
+    if (catalogApi) {
+      const retrievedEntity = await catalogApi.getEntityByRef(entityRef);
+      if (!retrievedEntity) {
+        throw new Error(`${entityRef.name} was not found`);
+      }
+      return retrievedEntity;
+    }
+    return undefined;
+  }, [apiHolder, entityRef]);
+
+  useEffect(() => {
+    if (popupState.isOpen && !entity && !error && !loading) {
+      load();
+    }
+  }, [popupState.isOpen, load, entity, error, loading]);
+
+  return (
+    <HoverPopover
+      PaperProps={{
+        className: classes.popoverPaper,
+      }}
+      {...bindPopover(popupState)}
+      anchorOrigin={{
+        vertical: 'bottom',
+        horizontal: 'center',
+      }}
+      transformOrigin={{
+        vertical: 'top',
+        horizontal: 'center',
+      }}
+    >
+      <Card>
+        {loading && <Progress />}
+        <CardContent>
+          <Typography color="textSecondary">{entityRef.namespace}</Typography>
+          <Typography variant="h5" component="div">
+            {entityRef.name}
+          </Typography>
+          {error && <Alert severity="warning">{error.message}</Alert>}
+          {entity ? (
+            <>
+              <Typography color="textSecondary">{entity.kind}</Typography>
+              <Typography paragraph>{entity.metadata.description}</Typography>
+              <Typography>{entity.spec?.type}</Typography>
```

Comment:
I think tags would also be nice to show

```suggestion
              <Typography>{entity.spec?.tags.join(', ')}</Typography>
```

Reply:
How many tags might there be on an entity?

Reply:
How many tags might there be on an entity?

Reply:
Thats a good question, so you think it should be capped? We usually have 1-4, but I guess it could be any number of at tags

```tsx
 <Typography>{_.take(entity.spec?.tags, 5).join(', ')}</Typography>
```

https://lodash.com/docs/4.17.15#take

Reply:
But description can also be any kind of length, right?

Reply:
Good point on the description. We should limit that.

Reply:
<img width="537" alt="image" src="https://user-images.githubusercontent.com/553697/206310171-9d6d50a5-3b89-4290-91e0-a531a059f205.png">

Reply:
Description is limited in length and Ive added the chips. If there are more than 4, it will add a chip with the text "..." and a tooltip.

Reply:
Looking good! FYI I am going to use it on a custom page with an embedded `EntityRelationsGraph` where I will add the peak component to the `renderNode` prop.

Reply:
Sounds interesting! :)